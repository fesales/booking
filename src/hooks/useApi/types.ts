export enum FetchStatus {
  ERROR = 'ERROR',
  PENDING = 'PENDING',
  SUCCESS = 'SUCCESS',
}

export interface FetchError {
  message: string;
}

interface PendingState {
  response: null;
  status: FetchStatus.PENDING;
}

export interface SuccessState<ApiResponse> {
  response: ApiResponse | null;
  status: FetchStatus.SUCCESS;
}

interface ErrorState {
  response: FetchError;
  status: FetchStatus.ERROR;
}

export type State<ApiResponse> = PendingState | SuccessState<ApiResponse> | ErrorState;

export enum Actions {
  REQUEST_LOADING = 'REQUEST_LOADING',
  REQUEST_SUCCESS = 'REQUEST_SUCCESS',
  REQUEST_ERROR = 'REQUEST_ERROR',
}

export interface RequestLoadingAction {
  type: Actions.REQUEST_LOADING;
}

export interface RequestSuccessAction<ApiResponse> {
  payload: ApiResponse;
  type: Actions.REQUEST_SUCCESS;
}

export interface RequestErrorAction {
  payload: FetchError;
  type: Actions.REQUEST_ERROR;
}

export type Action<ApiResponse> =
  | RequestErrorAction
  | RequestSuccessAction<ApiResponse>
  | RequestLoadingAction;
