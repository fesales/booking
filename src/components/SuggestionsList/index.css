.suggestions-list {
  margin: 0;
  background: #fff;
  border-radius: 4px;
  list-style: none;
  padding: 0;
  -webkit-box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
}

.suggestions-list-item {
  border-bottom: 1px solid #b4b4b4;
  display: flex;
  font-size: 12px;
  padding: 8px 16px;
}

.suggestions-list-item-name {
  display: block;
  font-size: 14px;
}

.suggestion-list-item-place {
  color: rgb(255, 255, 255);
}

.suggestion-list-item-place--A {
  background-color: rgb(150, 20, 18);
}

.suggestion-list-item-place--C {
  background-color: rgb(10, 99, 176);
}

.suggestion-list-item-place--T {
  background-color: rgb(91, 91, 91);
}
