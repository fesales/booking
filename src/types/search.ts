export enum PlaceType {
  AIRPORT = 'A',
  CITY = 'C',
  STATION = 'T',
}

export interface PickUpLocation {
  city: string;
  country: string;
  name: string;
  region: string;
  placeType: PlaceType;
}

export interface SearchApiResponse {
  results: {
    isGooglePowered: boolean;
    docs: PickUpLocation[];
    numFound: number;
  };
}
