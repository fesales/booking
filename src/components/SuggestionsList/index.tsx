import React from 'react';
import './index.css';
import { PickUpLocation } from '../../types/search';
import SuggestionsListItem from '../SuggestionsListItem';

interface Props {
  suggestions: PickUpLocation[];
}

const SuggestionsList: React.FunctionComponent<Props> = ({ suggestions }: Props) => {
  return (
    <ul className="suggestions-list">
      {suggestions.map(suggestion => (
        <SuggestionsListItem key={suggestion.name} {...suggestion} />
      ))}
    </ul>
  );
};

export default SuggestionsList;
