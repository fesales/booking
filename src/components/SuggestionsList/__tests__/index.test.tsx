import React from 'react';
import { shallow } from 'enzyme';
import SuggestionsList from '..';
import { PlaceType } from '../../../types/search';

describe('SuggestionsList', () => {
  it('should render component', () => {
    const props = {
      suggestions: [
        {
          city: 'Manchester',
          country: 'UK',
          name: 'pick up location',
          region: 'region',
          placeType: PlaceType.AIRPORT,
        },
      ],
    };

    const wrapper = shallow(<SuggestionsList {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('SuggestionsListItem')).toHaveLength(1);
  });
});
