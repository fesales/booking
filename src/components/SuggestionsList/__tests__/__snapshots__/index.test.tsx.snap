// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionsList should render component 1`] = `
<ul
  className="suggestions-list"
>
  <SuggestionsListItem
    city="Manchester"
    country="UK"
    key="pick up location"
    name="pick up location"
    placeType="A"
    region="region"
  />
</ul>
`;
